packages/renderer/src/stores/build-images.ts

```diff
@@ -16,13 +16,51 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
+import type { Terminal } from '@xterm/xterm';
 import type { Writable } from 'svelte/store';
-import { writable } from 'svelte/store';
+import { get, writable } from 'svelte/store';
+
+import type { ProviderContainerConnectionInfo } from '/@api/provider-info';
+
+export interface BuildArg {
+  key: string;
+  value: string;
+}
 
 export interface BuildImageInfo {
-  buildImageKey: symbol;
+  buildImageKey?: symbol;
   buildRunning: boolean;
+  buildFinished: boolean;
+  containerImageName: string;
+  containerFilePath: string;
+  containerBuildContextDirectory: string;
+  containerBuildPlatform: string;
+  buildParentImageName?: string;
+  buildError?: string;
+  buildArgs: BuildArg[];
+  cancellableTokenId?: number;
+  selectedProvider?: ProviderContainerConnectionInfo;
+  logsTerminal?: Terminal;
+  taskId: number;
+}
+
+export function cleanupBuildImageInfo(taskId: number): void {
+  const map = get(buildImagesInfo);
+  map.delete(taskId);
+  buildImagesInfo.set(map);
+}
+
+export function createDefaultBuildImageInfo(): BuildImageInfo {
+  return {
+    taskId: get(buildImagesInfo).size + 1,
+    containerImageName: '',
+    buildRunning: false,
+    buildFinished: false,
+    containerFilePath: '',
+    containerBuildContextDirectory: '',
+    containerBuildPlatform: '',
+    buildArgs: [{ key: '', value: '' }],
+  };
 }
 
-// current build key
-export const buildImagesInfo: Writable<BuildImageInfo> = writable();
+export const buildImagesInfo: Writable<Map<number, BuildImageInfo>> = writable(new Map());
```

Comment:
I don't know how future proof this is, I don't want to block this PR, because this is already how we do things in for the providers form etc.

But this is a memory leak, we never cleanup, we just store the data, if someone never restart podman desktop the build logs are stored forever, this is a problem..

 maybe we need some discussion and maybe think again about https://github.com/podman-desktop/podman-desktop/issues/7397

cc @benoitf, I am interested about your opinion on this matter

Reply:
This should be stored until it is accessible from Task Manager, so if someone decide to check what happened it specific build it would always get build image form restored to the saved state. 

I was going to implement clean up it up when Task removed from Task Manager and therefore the stored data for the build is no longer accessible from Podman Desktop UI.

Reply:
> I was going to implement clean up it up when Task removed from Task Manager and therefore the stored data for the build is no longer accessible from Podman Desktop UI.

yeah, but my point would be, maybe this should not be stored in the frontend, but in the backend, I don't know what is the best practice here.

If the data is stored in the backend, we could add a `cleanup / dispose` event to  the tasks, so when the tasks is cleaned / removed, we can removed attached data.